lib/chooseTeam.test.js

```diff
@@ -0,0 +1,157 @@
+    const expected = "X";
+
+    rl.question = (question, cb) => {
+      cb(expected);
+    };
+
+    const result = await getTeamChoice();
+
+    expect(result).toEqual(expected);
+  });
+
+  it("getTeamChoice returns user input in upperCase", async () => {
+    const expected = "X";
+    const input = "x";
+
+    rl.question = (question, cb) => {
+      cb(input);
+    };
+
+    const result = await getTeamChoice();
+    expect(result).toEqual(expected);
+  });
+
+  it("getValidTeamChoice returns true", () => {
+    const expected = true;
+    const input1 = "X";
+    const input2 = "O";
+
+    const result1 = getValidTeamChoice(input1);
+    const result2 = getValidTeamChoice(input2);
+    expect(result1).toEqual(expected);
+    expect(result2).toEqual(expected);
+  });
+
+  it("getValidTeamChoice returns false", () => {
+    const expected = false;
+    const input = "foo";
+
+    const result = getValidTeamChoice(input);
+    expect(result).toEqual(expected);
+  });
+
+  it("getValidTeamChoice validates getTeamChoice of 'X'", async () => {
+    const expected = true;
+    const input = "X";
+    rl.question = (question, cb) => {
+      cb(input);
+    };
+    const result = getValidTeamChoice(await getTeamChoice());
+    expect(result).toEqual(expected);
+  });
+
+  it("getValidTeamChoice validates getTeamChoice of 'foo'", async () => {
+    const expected = false;
+    const input = "foo";
+    rl.question = (question, cb) => {
+      cb(input);
+    };
+    const result = getValidTeamChoice(await getTeamChoice());
+    expect(result).toEqual(expected);
+  });
+
+  it("setTeam stores team choice of 'X'", () => {
+    const store = {
+      userTeam: null,
+      cpuTeam: null,
+    };
+    const expected = {
+      userTeam: "X",
+      cpuTeam: "O",
+    };
+    setTeam(store, "X");
+    expect(store).toEqual(expected);
+  });
+
+  it("setTeam stores team choice of 'O'", () => {
+    const store = {
+      userTeam: null,
+      cpuTeam: null,
+    };
+    const expected = {
+      userTeam: "O",
+      cpuTeam: "X",
+    };
+    setTeam(store, "O");
+    expect(store).toEqual(expected);
+  });
+
+  it("storeTeamChoice validates and stores choice of 'X'", async () => {
+    const store = {
+      userTeam: null,
+      cpuTeam: null,
+    };
+    const expected = {
+      userTeam: "X",
+      cpuTeam: "O",
+    };
+    const input = "X";
+
+    rl.question = (question, cb) => {
+      cb(input);
+    };
+
+    await storeTeamChoice(store);
+    expect(store).toEqual(expected);
+  });
+
+  it("storeTeamChoice validates and rejects choice of 'foo'", async () => {
+    const store = {
+      userTeam: null,
+      cpuTeam: null,
+    };
+    const expected = {
+      userTeam: "X",
+      cpuTeam: "O",
+    };
+    const expectedLog =
+      "Please enter only the letter O or X to choose your team";
+    const input = "foo";
+
+    rl.question = (question, cb) => {
+      cb(input);
+    };
```

Comment:
```suggestion
        rl.question: jest.fn()
            .mockImplementationOnce((questionText, cb) => {cb('foo')})
            .mockImplementationOnce((questionText, cb) => {cb('x')})
```

and then at the end of the test, do:

```
expect(rl.question.mock.calls.length).toBe(2);

```

Reply:
(following  https://jestjs.io/docs/en/mock-functions)

Reply:
or indeed https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber => `expect(rl.question). toHaveBeenCalledTimes(2);` (which is the same thing but prettier I guess)